core/src/components/input-otp/input-otp.tsx

```diff
@@ -0,0 +1,782 @@
+        this.el
+      );
+    }
+
+    const invalidSeparators = separatorValues.filter((pos) => pos > length);
+    if (invalidSeparators.length > 0) {
+      printIonWarning(
+        `[ion-input-otp] - The following separator positions are greater than the input length (${length}): ${invalidSeparators.join(
+          ', '
+        )}. These separators will be ignored.`,
+        this.el
+      );
+    }
+
+    this.parsedSeparators = separatorValues.filter((pos) => pos <= length);
+  }
+
+  componentWillLoad() {
+    this.processSeparators();
+    this.initializeValues();
+  }
+
+  /**
+   * Initializes the input values array based on the current value prop.
+   * This splits the value into individual characters and validates them against
+   * the allowed pattern. The values are then used as the values in the native
+   * input boxes and the value of the input group is updated.
+   */
+  private initializeValues() {
+    if (this.value != null && String(this.value).length > 0) {
+      const chars = String(this.value).split('').slice(0, this.length);
+      chars.forEach((char, index) => {
+        if (this.validKeyPattern.test(char)) {
+          this.inputValues[index] = char;
+        }
+      });
+      // Update the value without emitting events
+      this.value = this.inputValues.join('');
+    }
+  }
+
+  /**
+   * Resets the input values and focus state.
+   */
+  @Method()
+  async reset() {
+    this.inputValues = Array(this.length).fill('');
+    this.value = '';
+
+    this.focusedValue = null;
+    this.hasFocus = false;
+
+    this.inputRefs.forEach((input) => {
+      input.blur();
+    });
+
+    this.updateTabIndexes();
+  }
+
+  /**
+   * Sets focus to an input box.
+   * @param index - The index of the input box to focus (0-based).
+   * If provided and the input box has a value, the input box at that index will be focused.
+   * Otherwise, the first empty input box or the last input if all are filled will be focused.
+   */
+  @Method()
+  async setFocus(index?: number) {
+    if (typeof index === 'number') {
+      const validIndex = Math.max(0, Math.min(index, this.length - 1));
+      this.inputRefs[validIndex]?.focus();
+    } else {
+      const tabbableIndex = this.getTabbableIndex();
+      this.inputRefs[tabbableIndex]?.focus();
+    }
+  }
+
+  /**
+   * Get the regex pattern for allowed characters.
+   * If a pattern is provided, use it to create a regex pattern
+   * Otherwise, use the default regex pattern based on type
+   */
+  private get validKeyPattern(): RegExp {
+    return new RegExp(`^${this.getPattern()}$`, 'u');
+  }
+
+  /**
+   * Gets the string pattern to pass to the input element
+   * and use in the regex for allowed characters.
+   */
+  private getPattern(): string {
+    const { pattern, type } = this;
+    if (pattern) {
+      return pattern;
+    }
+    return type === 'number' ? '[0-9]' : '[\\p{L}\\p{N}]';
```

Comment:
```suggestion
    return type === 'number' ? '[\p{N}]' : '[\\p{L}\\p{N}]';
```
Very close! Missed this one though, which should allow for non-latic numbers with the number filter.

Reply:
https://github.com/ionic-team/ionic-framework/pull/30386/commits/1c9c939eef9a2c522f82d77cf98543b2fc45dc5d